src/main/java/org/skriptlang/skript/bukkit/spawner/elements/expressions/spawnrule/ExprSpawnRuleSkyLight.java

```diff
@@ -0,0 +1,135 @@
+package org.skriptlang.skript.bukkit.spawner.elements.expressions.spawnrule;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.spawner.SpawnRule;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.spawner.SpawnerModule;
+
+@Name("Spawn Rule - Sky Light Spawn Level")
+@Description({
+	"Returns the minimum/maximum sky light spawn levels of a spawn rule. "
+		+ "The sky light spawn levels determine the light level of the sky "
+		+ "that the spawner entry will spawn entities.",
+	"Note that the sky light spawn levels must be between 0 and 15, "
+		+ "the minimum sky light spawn level must be less than or equal to "
+		+ "the maximum sky light spawn level and vice versa."
+})
+@Examples({
+	"set {_entry} to a spawner entry using entity snapshot of a zombie:",
+		"\tset the weight to 5",
+		"\tset the spawn rule to a spawn rule:",
+			"\t\tset the minimum block light spawn level to 10",
+			"\t\tset the maximum block light spawn level to 15",
+			"\t\tset the minimum sky light spawn level to 5",
+			"\t\tset the maximum sky light spawn level to 15",
+	"set spawner entity of event-block to {_entry}"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Minecraft 1.21+")
+public class ExprSpawnRuleSkyLight extends SimplePropertyExpression<SpawnRule, Integer> {
+
+	static {
+		registerDefault(SpawnerModule.SYNTAX_REGISTRY, ExprSpawnRuleSkyLight.class, Integer.class,
+				"(1:max[imum]|min[imum]) sky[ ]light [entity] spawn [rule] (level|value)", "spawnrules"
```

Comment:
to clean this up, couldnt you do
`(1:max|min)[imum]`?
same with the previous class